### DISCUSSION TITLE
How to wait for a container with healtcheck?

### DISCUSSION THREAD

**Discussion by kfrajtak** [testcontainers/testcontainers-dotnet#531] [Q&A] [answered]
I have a container that takes some time to spin up and health check to determine when container is ready:

How to wait for the healthy state?

```
FROM mcr.microsoft.com/mssql/server:2022-latest

ENV ACCEPT_EULA=Y
ENV SA_PASSWORD=Password123
ENV LD_PRELOAD=/nodirect_open.so

USER root
RUN wget https://raw.githubusercontent.com/t-oster/mssql-docker-zfs/master/nodirect_open.c
RUN apt update
RUN apt install -y gcc
RUN gcc -shared -fpic -o /nodirect_open.so nodirect_open.c -ldl
RUN apt purge -y gcc
RUN apt clean
RUN echo "/nodirect_open.so" >> /etc/ld.so.preload

HEALTHCHECK --interval=10s --timeout=5s --start-period=10s --retries=10 \
CMD /opt/mssql-tools/bin/sqlcmd -S . -U sa -P Password123 -Q "SELECT \"READY\"" | grep -q "READY"
```

Thanks

**Comment by HofmeisterAn** [comment, marked as answer]
It's not supported yet, but a contribution would be nice.

**Reply by kfrajtak** [reply]
What is the process if I want to contribute? As described in the contribution guide - just fork and branch?

**Reply by HofmeisterAn** [reply]
Yes, fork the repository, create a feature branch and a pull request. Consider the code style and tests.

**Comment by kfrajtak** [comment]
@HofmeisterAn where did the `ITestcontainersContainer.Health` property go?

**Reply by HofmeisterAn** [reply]
The status has not changed (same as Jul 30), it is not implemented. If you think about a contribution, please take a look at: [Contributing](https://github.com/testcontainers/testcontainers-dotnet/blob/develop/CONTRIBUTING.md).

**Reply by kfrajtak** [reply]
Sorry, my bad - bad synchronization between computers. My apologies.
